### ISSUE TITLE
add support for nested navigation

### ISSUE THREAD

**Issue by csells** [csells/go_router#4]
use https://github.com/flutter/samples/pull/832 as a reference example to implement

**Comment by csells** [comment]
work in progress, but have broken redirection (redirect and query params samples) and nested routing doesn't actually work. plus, providing a Navigator by hand is just ugly even if it did work...

**Comment by csells** [comment]
Do I really need this?

```dart
Iterable<GoRoute> _routes(BuildContext context, String location) {
  final library = context.read<Library>();
  return [
    GoRoute(
      pattern: '/books',
      builder: (context, state) => MaterialPage<BooksScreen>(
        key: state.pageKey,
        child: BooksScreen(tab: state.params['nested']!, child: state.child!),
      ),
      nestedRoutes: [
        GoNestedRoute(
          pattern: 'popular',
          builder: (context, state) => BookList(books: library.popularBooks),
        ),
        GoNestedRoute(
          pattern: 'new',
          builder: (context, state) => BookList(books: library.newBooks),
        ),
        GoNestedRoute(
          pattern: 'all',
          builder: (context, state) => BookList(books: library.allBooks),
        ),
      ],
    ),
  ];
}
```

**Comment by csells** [comment]
because w/ [real nested navigation](https://github.com/csells/just_books), you get transitions for just the bits that have changed and not the whole page.

**Comment by csells** [comment]
fixed with v0.8.0.
